billing/actions/accounts.py

```diff
@@ -220,3 +228,123 @@ def assign_funds_to_invoice(invoice_id: str) -> bool:
             Charge.objects.create(account_id=account_id, amount=-overpayment, product_code=CREDIT_REMAINING)
 
     return invoice.status == Invoice.PAID
+
+
```

Comment:
Not certain about this but mark_accounts_as_delinquent and mark_accounts_as_legal are very similar so there is a lot of duplicated code/logic. Also, when you call them from the management command you call both.
Maybe it would be interesting to just have one function that updates the delinquent status (from compliant to delinquent and also from delinquent to compliant)?

Reply:
I join them in just one called `update_accounts_delinquent_status`